src/components/ActiveDatesCard/ActiveDatesCard.tsx

```diff
@@ -57,9 +57,11 @@ export function ActiveDatesCard({
 
   const ianaTimezone = i18n.defaultTimezone!;
 
-  const [showEndDate, setShowEndDate] = useState(
-    initShowEndDate(startDate, endDate),
-  );
+  const [showEndDate, setShowEndDate] = useState(Boolean(endDate.value));
+
+  useEffect(() => {
+    setShowEndDate(Boolean(endDate.value));
+  }, [endDate.value]);
```

Comment:
[question] We might be able to save a rerender if we don't use the `hooks`.  I think it should result in the same thing, as the `setShowEndDate` is only ever called on `handleShowEndDateChange` (which is a checkbox to change the `endDate` value) and in this `useEffect`.  Is there something I am missing
```suggestion
  const showEndDate = Boolean(endDate.value);
```

Reply:
Good question, let check 🤔

Reply:
Nice catch! I played around with it in storybook and it works fine without the hooks